platform/mbed_wait_api.h

```diff
@@ -55,9 +55,9 @@ extern "C" {
  *  @param s number of seconds to wait
  *
  *  @note
- *    If the RTOS is present, this function always spins to get the exact number of microseconds,
- *    which potentially affects power (such as preventing deep sleep) and multithread performance.
- *    You can avoid it by using Thread::wait().
+ *    If the RTOS is present, this function spins to get the exact number of microseconds for 
+ *    usec precision upto 10msec. If delay is larger then 10msec and not in ISR, it is same as 
+ *    `wait_ms`. `wait_us` and `wait_ms` are recommended over `wait`
```

Comment:
According to Kevin's comment that function should be deprecated. 
Question: should we have a `wait_s` function that accepts an integral type ?

Reply:
@kjbracey-arm  - Still work for `wait` will be needed in future and I would like to see `sleep_for` and `sleep_until` replacing `wait`. Shall we keep deprecation for next level of improvement on wait?